contrib/dockerd-rootless.sh

```diff
@@ -105,5 +105,12 @@ else
 	# remove the symlinks for the existing files in the parent namespace if any,
 	# so that we can create our own files in our mount namespace.
 	rm -f /run/docker /run/containerd /run/xtables.lock
+
+	# Set XTABLES_LOCKFILE (default: "/run/xtables.lock") to a custom path under /tmp,
+	# to avoid hitting issue moby/moby#41230 on SELinux-enabled hosts.
+	#
+	# XTABLES_LOCKFILE is recognized since iptables v1.8.6.
+	XTABLES_LOCKFILE="$(mktemp /tmp/xtables.XXXXXXXX)"
```

Comment:
Note: I also tried `/run/dockerd-rootless-xtables.lock`, `$(mktemp -d /tmp/dir.XXXXXXXX)/xtables.lock`, `${ROOTLESSKIT_STATE_DIR/xtables.lock}`, `$HOME/.docker/run/xtables.lock`, but none of them could satisfy the SELinux cargo god 😅 

It seems only a file under `/tmp` can work.

Reply:
@giuseppe Do you happen to know whether this is by design?

Reply:
no idea :)  what error do you see?

Could you force a `system_u:object_r:iptables_var_run_t:s0` label on the file?

Reply:
`can't open lock file /run/user/1000/dockerd-rootless-xtables.lock: Permission denied error`.

I tried `touch  /run/user/1000/dockerd-rootless-xtables.lock && chcon system_u:object_r:iptables_var_run_t:s0  /run/user/1000/dockerd-rootless-xtables.lock` but same error

Reply:
Updated to use `chcon system_u:object_r:iptables_var_run_t:s0 /run`.